src/lib/encoder.ts

```diff
@@ -60,30 +57,33 @@ const bytesNRegex = /Bytes(\d+)/;
 const ALLOWED_BYTES_SIZES = [2, 4, 8, 16, 32, 64, 128, 256];
 
 const encodeDataSourceWithHash = (
-  hashType: SUPPORTED_VERIFICATION_FUNCTIONS,
-  dataHash: string,
+  verification: undefined | Verification,
   dataSource: string,
 ): string => {
-  const verificationFunction = getVerificationFunction(hashType);
-
+  if (!verification || !verification.method || !verification.data) {
+    throw new TypeError('No verification function or data provided.');
```

Comment:
do we definitely want to throw here?

Reply:
Changed it to support a 0000 or unknown verification method. This will always verify as good.